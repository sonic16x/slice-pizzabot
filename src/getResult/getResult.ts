import { parseUserInput } from '../parseUserInput';
import { getPath } from '../getPath';
import { generateSvg } from '../generateSvg';

interface IResult {
    path: string;
    svg: string;
    error: string;
}

export function getResult(inputParams: string): IResult {
    try {
        const { size, targets } = parseUserInput(inputParams);

        const path = getPath(targets);
        const svg = generateSvg(size, targets, path);

        return {
            path,
            svg,
            error: '',
        };
    } catch ({ message }) {
        return {
            path: '',
            svg: '',
            error: message as string,
        };
    }
}
