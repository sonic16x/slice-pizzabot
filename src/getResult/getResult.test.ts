import { getResult } from './getResult';

test('getResult test', () => {
    expect(getResult('5x5 (0, 0) (1, 3) (4, 4) (4, 2) (4, 2) (0, 1) (3, 2) (2, 3) (4, 1)')).toMatchSnapshot();
    expect(getResult('asdasd')).toMatchSnapshot();
});
