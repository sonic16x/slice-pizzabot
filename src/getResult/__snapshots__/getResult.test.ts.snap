// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getResult test 1`] = `
Object {
  "error": "",
  "path": "DNDENNDEDESDEDDSDNNND",
  "svg": "
        <svg width=\\"350\\" height=\\"350\\" xmlns=\\"http://www.w3.org/2000/svg\\">
            <rect width=\\"100%\\" height=\\"100%\\" fill=\\"#fff\\" stroke=\\"black\\" stroke-width=\\"2\\"/>
            <line x1=\\"50\\" y1=\\"0\\" x2=\\"50\\" y2=\\"350\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"100\\" y1=\\"0\\" x2=\\"100\\" y2=\\"350\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"150\\" y1=\\"0\\" x2=\\"150\\" y2=\\"350\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"200\\" y1=\\"0\\" x2=\\"200\\" y2=\\"350\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"250\\" y1=\\"0\\" x2=\\"250\\" y2=\\"350\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"300\\" y1=\\"0\\" x2=\\"300\\" y2=\\"350\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"0\\" y1=\\"300\\" x2=\\"350\\" y2=\\"300\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"0\\" y1=\\"250\\" x2=\\"350\\" y2=\\"250\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"0\\" y1=\\"200\\" x2=\\"350\\" y2=\\"200\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"0\\" y1=\\"150\\" x2=\\"350\\" y2=\\"150\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"0\\" y1=\\"100\\" x2=\\"350\\" y2=\\"100\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
<line x1=\\"0\\" y1=\\"50\\" x2=\\"350\\" y2=\\"50\\" style=\\"stroke:rgb(50,50,50);stroke-width:1\\" />
            <polygon points=\\"50,0 55,5 45,5\\" style=\\"fill:rgb(50,50,50);stroke-width:1\\" />
<polygon points=\\"350,300 345,295 345,305\\" style=\\"fill:rgb(50,50,50);stroke-width:1\\" />
            <line x1=\\"50\\" y1=\\"300\\" x2=\\"50\\" y2=\\"250\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"50\\" y1=\\"250\\" x2=\\"100\\" y2=\\"250\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"100\\" y1=\\"250\\" x2=\\"100\\" y2=\\"200\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"100\\" y1=\\"200\\" x2=\\"100\\" y2=\\"150\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"100\\" y1=\\"150\\" x2=\\"150\\" y2=\\"150\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"150\\" y1=\\"150\\" x2=\\"200\\" y2=\\"150\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"200\\" y1=\\"150\\" x2=\\"200\\" y2=\\"200\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"200\\" y1=\\"200\\" x2=\\"250\\" y2=\\"200\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"250\\" y1=\\"200\\" x2=\\"250\\" y2=\\"250\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"250\\" y1=\\"250\\" x2=\\"250\\" y2=\\"200\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"250\\" y1=\\"200\\" x2=\\"250\\" y2=\\"150\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
<line x1=\\"250\\" y1=\\"150\\" x2=\\"250\\" y2=\\"100\\" style=\\"stroke-dasharray: 5 5 ; stroke: #00F; stroke-width: 2;\\" />
            <circle cx=\\"50\\" cy=\\"300\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"100\\" cy=\\"150\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"250\\" cy=\\"100\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"250\\" cy=\\"200\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"250\\" cy=\\"200\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"50\\" cy=\\"250\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"200\\" cy=\\"200\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"150\\" cy=\\"150\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
<circle cx=\\"250\\" cy=\\"250\\" r=\\"5\\" stroke=\\"#f00\\" stroke-width=\\"0\\" fill=\\"#f00\\" />
            <text x=\\"55\\" y=\\"295\\" class=\\"small\\">(0, 0)</text>
<text x=\\"105\\" y=\\"145\\" class=\\"small\\">(1, 3)</text>
<text x=\\"255\\" y=\\"95\\" class=\\"small\\">(4, 4)</text>
<text x=\\"255\\" y=\\"195\\" class=\\"small\\">(4, 2)</text>
<text x=\\"255\\" y=\\"195\\" class=\\"small\\">(4, 2)</text>
<text x=\\"55\\" y=\\"245\\" class=\\"small\\">(0, 1)</text>
<text x=\\"205\\" y=\\"195\\" class=\\"small\\">(3, 2)</text>
<text x=\\"155\\" y=\\"145\\" class=\\"small\\">(2, 3)</text>
<text x=\\"255\\" y=\\"245\\" class=\\"small\\">(4, 1)</text>
<text x=\\"60\\" y=\\"320\\" class=\\"small\\">0</text>
<text x=\\"110\\" y=\\"320\\" class=\\"small\\">1</text>
<text x=\\"160\\" y=\\"320\\" class=\\"small\\">2</text>
<text x=\\"210\\" y=\\"320\\" class=\\"small\\">3</text>
<text x=\\"260\\" y=\\"320\\" class=\\"small\\">4</text>
<text x=\\"310\\" y=\\"320\\" class=\\"small\\">5</text>
<text x=\\"30\\" y=\\"290\\" class=\\"small\\">0</text>
<text x=\\"30\\" y=\\"240\\" class=\\"small\\">1</text>
<text x=\\"30\\" y=\\"190\\" class=\\"small\\">2</text>
<text x=\\"30\\" y=\\"140\\" class=\\"small\\">3</text>
<text x=\\"30\\" y=\\"90\\" class=\\"small\\">4</text>
<text x=\\"30\\" y=\\"40\\" class=\\"small\\">5</text>
        </svg>
    ",
}
`;

exports[`getResult test 2`] = `
Object {
  "error": "Incorrect input: use only integer numbers in field size",
  "path": "",
  "svg": "",
}
`;
